src/components/manifold-subscription-create/manifold-subscription-create.tsx

```diff
@@ -196,15 +226,21 @@ export class ManifoldSubscriptionCreate {
       this.calculatedCost = cost;
     } catch (e) {
       if (e.name !== 'AbortError') {
-        // TODO store error in a better way so it can be displayed in place of cost in UI
-        this.errors = [{ message: 'Error getting plan cost.' }];
+        this.addErrors(dataError('cost', 'cost of selected plan'));
       }
     }
     return undefined;
   }
 
   subscribe = async (e: UIEvent) => {
     e.preventDefault();
+    this.removeErrors('subscription');
+
+    if (!this.connection) {
```

Comment:
I've been taking a... slightly different angle on this. Maybe you'll think this is cheating. For starters, I don't make the `connection` optional.

https://github.com/manifoldco/manifold-subscription/blob/master/src/components/manifold-subscription-details/manifold-subscription-details.tsx#L14

It's just uninitialized. And I assume I can get away with it because I `await core.initialize` in the `componentWillLoad` function, so I know that the connection will be there before any of my component code runs.

I think the obvious downside is that a developer could break this. But we could also quite possibly lint for it.

Reply:
yeah, I think that's fair. the component won't load without the connection. I wonder if there should be a timeout (in componentWillLoad) and we throw an error if the connection never comes. there might be a situation with for whatever reason manifold-init doesn't get added to the dom?